kernel/safe_typing.ml

```diff
@@ -1351,14 +1286,93 @@ let import lib cst vodigest senv =
     Modops.add_linked_module mb linkinfo env
   in
   let env = Environ.add_native_symbols lib.comp_name lib.comp_natsymbs env in
+  let sections =
+    Option.map (Section.map_custom (fun custom ->
+        {custom with rev_reimport = (lib,cst,vodigest) :: custom.rev_reimport}))
```

Comment:
For clarity, shouldn't we share the declaration of universe constraints with the mechanism already there for sections? That is, when replaying requires, we shouldn't push any constraints. If so, the code below needs to be adapted by switching replaying of global constraints and requires.

Reply:
I don't see how that's clearer.

Reply:
Well, at least we don't have *two* mechanisms registering monomorphic constraints. This is definitely asking for trouble, even if currently correct.

Now that I think of it, the current PR might even be unsound. What happens if you require inside a polymorphic section a carefully crafted module that refers to local universe variables?